### PR TITLE
Add dummy parts

### PR DESCRIPTION
**Pull request by mehdisadeghi**
This PR adds a new CLI key `--dummy` which will produce the following files with dummy data:

 - `calendar.txt`
 - `trips.txt`
 - `stop_times.txt`

It also adds by default a new file `shapes.txt`. 

@carocad since this PR is getting too large there might be other missing parts that unless necessary I will address in separate PRs. I have also chosen a rather silly timing schedule, If it doesn't work let me know of a better schedule and I'll replace it.

### CODE REVIEW

File: tests/out/shapes.txt [hunk 1]
```diff
@@ -0,0 +1,1774 @@
+shape_id,shape_pt_lat,shape_pt_lon,shape_pt_sequence
+shp_19914,53.0507001,8.7896481,0
```
Comment (carocad) [inline tests/out/shapes.txt]: @mehdisadeghi I guess that you decided to use the "shp" prefix to differentiate the use of the same relation id for several output files. I think this can lead to some problems down the road since the use of that prefix means that we need to treat those fields as strings not as integers anymore which greatly impact the memory consumption. I would propose the following alternatives:
- use the same id for shape_id, trip_id, etc. Since those fields are tagged directly in the files, there shouldnt be any problem in using the same id for several fields as long as they have a different tag.
- use an increment counter for each id
- use a concatenation of relation id and changeset number. Both of those fields are integers so concatenating the two values should also produce an integer value.

what do you think?

File: tests/out/trips.txt [hunk 2]
```diff
@@ -0,0 +1,627 @@
+route_id,service_id,trip_id,trip_headsign,shape_id
+19914,WE,trp_19914,Dummy Trip,shp_19914
```
Comment (carocad) [inline tests/out/trips.txt]: @mehdisadeghi would it be possible to have the relation "name" as the headsign? This can obviously be handled in another PR but I just wanted to ask it here since the context is pretty useful ;)

File: tests/tests_profile.txt [hunk 3]
```diff
@@ -1,2684 +0,0 @@
-============================= test session starts ==============================
```
Comment (carocad) [inline tests/tests_profile.txt]: @mehdisadeghi why did this file got deleted? accident?

File: tests/benchmark.txt [hunk 4]
```diff
@@ -1,1254 +1,1269 @@
 GTFS feed saved in /home/mehdi/Projects/hiposfer/osmtogtfs/tests/out
-         7526061 function calls (7523741 primitive calls) in 23.256 seconds
+         3751892 function calls (3749564 primitive calls) in 16.397 seconds
```
Comment (carocad) [inline tests/benchmark.txt]: @mehdisadeghi nice improvement :)

really nice to be able to identify it inmediately

File: tests/out/trips.txt (hunk 2 above)
Comment (mehdisadeghi) [reply tests/out/trips.txt]: @carocad hmm. Yes of course, however I wanted to make it obvious that this is a dummy trip. Perhaps a combination of relation + dummy will do it.

File: tests/out/shapes.txt (hunk 1 above)
Comment (mehdisadeghi) [reply tests/out/shapes.txt]: @carocad Yes this can definitely be improved. The following points come to my mind:

 - GTFS description asks only for _dataset unique_ ids, not integers.
 - Any solution that we come up with should reproduce the same ids over different runs
 - concatenating ids and changesets sounds like a good idea!

File: tests/out/trips.txt (hunk 2 above)
Comment (carocad) [reply tests/out/trips.txt]: yeah I understand that differentiate those is important. I was thinking more in terms of the front-end. There we need to display something to the user such that it is clear which route/trip are we referring to. Obviously showing `Dummy trip` to a user would not be very useful. I think that just as you said, having a combination of real + dummy would be ideal. That way we can strip the dummy part at runtime and display only a nice headsign.

File: tests/tests_profile.txt (hunk 3 above)
Comment (mehdisadeghi) [reply tests/tests_profile.txt]: @carocad actually I removed it because it sounds like unnecessary to profile tests. Instead I focused on profiling only the script (which tests will use anyway).

File: tests/benchmark.txt (hunk 4 above)
Comment (mehdisadeghi) [reply tests/benchmark.txt]: @carocad yeah, actually this is a simple way that I saw you using in `kamal` so I followed the suit! :)
BTW, I realized that memory profiling in the script is as important as the speed tests. I was unable to process Hessen data due to insufficient memory. The main problem is the huge number of nodes which I have to store them for further processing. I digged a bit and realized Osmium has special support for some stores and high performance processing which I have yet to discover.

Comment (carocad) [reply tests/benchmark.txt]: yeah, this is why I switched from custom osm preprocessing to overpass-api query. By leaving the filtering outside of the scope I can focus on data processing instead of joggling with memory issues.

> BTW, I realized that memory profiling in the script is as important as the speed tests. I was unable to process Hessen data due to insufficient memory

File: tests/out/shapes.txt (hunk 1 above)
Comment (mehdisadeghi) [reply tests/out/shapes.txt]: @carocad I used relation id as shape id.

Comment (mehdisadeghi) [reply tests/out/shapes.txt]: @carocad btw, we shouldn't consider every gtfs file to have integers as IDs, because they don't.

**Comment by mehdisadeghi** [comment]
@carocad please see my latest changes.

File: tests/out/trips.txt [hunk 5]
```diff
@@ -1,627 +1,627 @@
...
-6830158,WD,trp_6830158,Dummy Trip,shp_6830158
-6864466,WE,trp_6864466,Dummy Trip,shp_6864466
-6864466,WD,trp_6864466,Dummy Trip,shp_6864466
-6925892,WE,trp_6925892,Dummy Trip,shp_6925892
-6925892,WD,trp_6925892,Dummy Trip,shp_6925892
-6925926,WE,trp_6925926,Dummy Trip,shp_6925926
-6925926,WD,trp_6925926,Dummy Trip,shp_6925926
-6926535,WE,trp_6926535,Dummy Trip,shp_6926535
-6926535,WD,trp_6926535,Dummy Trip,shp_6926535
+19914,WE,trp_19914,[Dummy]Flughafen-Süd-to-Universität-Nord,19914
```
Comment (carocad) [inline tests/out/trips.txt]: nice :)

File: tests/out/shapes.txt [hunk 6]
```diff
@@ -1,1774 +1,1774 @@
...
-6559183,53.1858632,8.6376531,4
-6559183,53.1885375,8.6406411,5
-6560472,53.1947326,8.6150382,0
-6560472,53.1925607,8.6171911,1
-6560472,53.190696,8.6184409,2
-6560472,53.1861406,8.6191371,3
-6560472,53.1700892,8.6279496,4
-6648747,53.1214837,8.7526593,0
-6785438,53.1684836,8.6615385,0
+19914,8.7896481,53.0507001,0
```
Comment (carocad) [inline tests/out/shapes.txt]: @mehdisadeghi the lat and lon are swapped comparing them with the previous output. If I have to guess I would say that the current one is wrong since you are using Germany's data which means lat around 50 and lon around 8

Comment (mehdisadeghi) [reply tests/out/shapes.txt]: @carocad oh wait let me see, too much refactoring today! 😆

Comment (mehdisadeghi) [reply tests/out/shapes.txt]: image

Yeah...

File: tests/out/stop_times.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+trip_id,arrival_time,departure_time,stop_id,stop_sequence
```
Comment (carocad) [inline tests/out/stop_times.txt]: @mehdisadeghi is this suppose to be empty? I think it was full some commits ago 🤔

Comment (mehdisadeghi) [reply tests/out/stop_times.txt]: > some commit ago

:D oh man... I have to take a break ;-)

Comment (carocad) [reply tests/out/stop_times.txt]: hahahha yeah I agree. The rest looks pretty good so just to prevent making this PR even bigger I would say to stop there.

Comment (mehdisadeghi) [reply tests/out/stop_times.txt]: @carocad I found the problem. I'll push a fix in a few minutes, I'm checking that. While fixing that I found a case related to multi-regional routes which you can find in the lengthy comment inside `gtfs_dummy.py` file.

**Comment by mehdisadeghi** [comment]
@carocad please see the latest changes and hopefully we close this tonight.

File: tests/out/stop_times.txt [hunk 8]
```diff
@@ -0,0 +1,1774 @@
...
+199141,05:00:00,05:00:30,773271753,0
+199141,05:00:00,05:00:30,773271757,1
```
Comment (carocad) [inline tests/out/stop_times.txt]: @mehdisadeghi for another PR, but I guess that this is a bug here. It says that both stop 0 and 1 are reached at the same time and depart at the same time. At the very least it should arrive after the departure of the previous one. or did I get this wrong?

Comment (mehdisadeghi) [reply tests/out/stop_times.txt]: Good catch @carocad! This is indeed a bug, I have made a mistake in my loop while generating this data.

Comment (mehdisadeghi) [reply tests/out/stop_times.txt]: @carocad I'm pushing a fix for that right now. It is a tiny.

**Comment by carocad** [comment]
@mehdisadeghi nice, looks pretty solid now :)
